miio/pet_water_dispenser.py

```diff
@@ -0,0 +1,274 @@
+
+    @property
+    def days_before_cleaning(self) -> int:
+        return self.data['remain_clean_time']
+
+    @property
+    def no_water(self) -> bool:
+        '''True if there is no water left'''
+        if self.data['no_water_flag']:
+            return False
+        return True
+
+    @property
+    def minutes_without_water(self) -> int:
+        return self.data['no_water_time']
+
+    @property
+    def pump_is_blocked(self) -> bool:
+        '''True if pump is blocked'''
+        return self.data['pump_block_flag']
+
+    @property
+    def lid_is_up(self) -> bool:
+        '''True if lid is up'''
+        return self.data['lid_up_flag']
+
+    @property
+    def timezone(self) -> int:
+        return self.data['timezone']
+
+    @property
+    def location(self) -> str:
+        return self.data['location']
+
+    @property
+    def error_detected(self) -> bool:
+        '''True if fault detected'''
+        return self.data['fault'] > 0
+
+
+class PetWaterDispenser(MiotDevice):
+    '''Main class representing the pet water dispenser.'''
+
+    mapping = _MAPPING
+
+    @command(
+        default_output=format_output(
+            '',
+            'Cotton filter live: {result.cotton_left_time} day(s) left\n'
+            'Days before cleaning: {result.days_before_cleaning} day(s)\n'
+            'Error detected: {result.error_detected}\n'
+            'Filter live: {result.filter_left_time} day(s) left\n'
+            'Indicator light enabled: {result.indicator_light_enabled}\n'
+            'Lid is up: {result.lid_is_up}\n'
+            'Location: {result.location}\n'
+            'Mode: {result.mode}\n'
+            'Minutes without water: {result.minutes_without_water} minute(s)\n'
+            'No water: {result.no_water}\n'
+            'Power: {result.power}\n'
+            'Pump is blocked: {result.pump_is_blocked}\n'
+            'Timezone: {result.timezone}\n'
+        )
+    )
+    def status(self) -> PetWaterDispenserStatus:
+        '''Retrieve properties.'''
+        return PetWaterDispenserStatus(
+            {
+                prop['did']: prop['value'] if prop['code'] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(default_output=format_output('Powering off ...'))
+    def off(self) -> Any:
+        '''Power off.'''
+        status = self.status()
+        if not status.is_on:
+            return 'Already off'
+        resp = self.set_property('on', False)
+        if resp[0]['code'] != 0:
+            raise PetWaterDispenserException(resp)
+        return 'ok'
+
+    @command(default_output=format_output('Powering on ...'))
+    def on(self) -> str:
+        '''Power on.'''
+        status = self.status()
+        if status.is_on:
+            return 'Already on'
+        resp = self.set_property('on', True)
+        if resp[0]['code'] != 0:
+            raise PetWaterDispenserException(resp)
+        return 'ok'
```

Comment:
```suggestion
        return self.set_property('on', True)
```
Just return the return value of the `set_property` on all cases, it's up to the caller to handle the error cases (for now, at least).